File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int maxi = 0;
        vector<bool>ans(candies.size(),true);
        for(int i = 0;i < candies.size();i ++){
            maxi = max(maxi,candies[i]);
        }
        int p = maxi - extraCandies;
        for(int i = 0;i < candies.size();i ++){
            if(candies[i] < p)ans[i] = false;
        }
        return ans;
    }
};

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        n = len(candies)
        maxi = max(candies)
        ans = []
        for i in range(n):
            if candies[i] >= maxi - extraCandies :
                ans.append(True)
            else :
                 ans.append(False)
        return ans
        
